pkg/gui/controllers/helpers/inline_status_helper.go

```diff
@@ -0,0 +1,129 @@
+package helpers
+
+import (
+	"time"
+
+	"github.com/jesseduffield/gocui"
+	"github.com/jesseduffield/lazygit/pkg/gui/types"
+	"github.com/jesseduffield/lazygit/pkg/utils"
+	"github.com/sasha-s/go-deadlock"
+)
+
+type InlineStatusHelper struct {
+	c *HelperCommon
+
+	contextsWithInlineStatus map[types.ContextKey]*inlineStatusInfo
```

Comment:
So far helpers have been stateless: letting the Gui or GuiRepoState struct hold any required state. I'm torn about whether we should retain that pattern here or just store the state on the helper. Having it on the helper is good for encapsulation but makes it harder to see at a glance the state of the app.

I'm happy to have it on the helper and see how that goes

Reply:
Hm, I'm torn too. We had this discussion elsewhere (can't find it right now) to maybe convert all helpers to freestanding functions, in order to make interdependencies between helpers more convenient. This would be the only reason for me to keep the state out of here. Unless we accept that some helpers stay objects.

But I really like the encapsulation here; the alternative would be to expose the `contextsWithInlineStatus` map publicly in GuiCommon, but it really isn't anybody else's business to mess with that state.

Reply:
@jesseduffield I was holding off of merging this one to see if there's any more input on this question. I'd like to merge now though, because it conflicts with #3058.

Reply:
Go ahead: happy to see how this approach goes